File: java_src/DH3DLibrary/src/jp/spec/darkhorse2/dh3d/mmd/PMDModel.java
/**
 * 
 */
package jp.spec.darkhorse2.dh3d.mmd;

import jp.spec.darkhorse2.dh3d.base.*;

/**
 * @author Norihiro
 *
 */
public class PMDModel extends Model{
	public float version;
	public String modelName;
	public String comment;
	public Face[] faceArray;
	public Face<String> faceHash;
	public int[] faceDisplayArray;
	public String[] boneDisplayNameArray;
	public int[] boneDisplayIndex;
	public int[] boneDisplayFrameIndex;
	public boolean englishCompatibility;
	public String englishName;
	public String englishComment;
	public String[] boneDisplayEnglishNameArray;
	
	
}

File: src/js/mmd/PMDModel.js
'use strict'

import Model from '../base/Model'

/**
 * PMDModel class
 * @access public
 */
export default class PMDModel extends Model {
  /**
   * constructor
   * @access public
   * @constructor
   */
  constructor() {
    super()

    // Header
    this.version = 0.0
    this.modelName = ''
    this.comment = ''

    // Face
    this.faceArray = []
    this.faceHash = new Map()

    // FaceDisplay
    this.faceDisplayArray = []

    // BoneDisplayName
    this.boneDisplayNameArray = []

    // BoneDisplay
    this.boneDisplayIndex = []
    this.boneDisplayFrameIndex = []

    // English
    this.englishCompatibility = false
    this.englishName = ''
    this.englishComment = ''
    this.boneDisplayEnglishNameArray = []

    // Motion
    this.motionNumCache = new Map()
    this.faceMotionNumCache = new Map()
  }

  destroy() {
    super.destroy()
  }

  copy(model) {
    super.copy(model)

    this.version                     = model.version
    this.modelName                   = model.modelName
    this.comment                     = model.comment
    this.faceArray                   = model.faceArray
    this.faceHash                    = model.faceHash
    this.faceDisplayArray            = model.faceDisplayArray
    this.boneDisplayNameArray        = model.boneDisplayNameArray
    this.boneDisplayIndex            = model.boneDisplayIndex
    this.boneDisplayFrameIndex       = model.boneDisplayFrameIndex
    this.englishCompatibility        = model.englishCompatibility
    this.englishName                 = model.englishName
    this.englishComment              = model.englishComment
    this.boneDisplayEnglishNameArray = model.boneDisplayEnglishNameArray
  }
}
